python/lsst/meas/algorithms/normalizedCalibrationFlux.py

```diff
@@ -0,0 +1,295 @@
+        raw_fluxerr_name = raw_name + "_instFluxErr"
+        norm_name = self.config.normalized_calibflux_name
+
+        if self.config.do_measure_ap_corr:
+            ap_corr_field, ap_corr_err_field = self._measure_aperture_correction(exposure, catalog)
+        else:
+            ap_corr_map = exposure.info.getApCorrMap()
+            if ap_corr_map is None:
+                raise RuntimeError("NormalizedCalibrationFluxTask is configured with "
+                                   "do_measure_ap_corr=False but exposure has no aperture correction map.")
+            ap_corr_field = ap_corr_map.get(raw_flux_name)
+            ap_corr_err_field = ap_corr_map.get(raw_fluxerr_name)
+            if not ap_corr_field or not ap_corr_err_field:
+                raise RuntimeError("NormalizedCalibrationFlux is configured with "
+                                   f"do_measure_ap_corr=False but {raw_flux_name}/{raw_fluxerr_name} "
+                                   "not in exposure aperture correction map.")
+
+        corrections = ap_corr_field.evaluate(
+            catalog["slot_Centroid_x"],
+            catalog["slot_Centroid_y"],
+        )
+
+        input_flux_name = raw_flux_name
+        input_fluxerr_name = raw_fluxerr_name
+        input_flag_name = raw_name + "_flag"
+        output_flux_name = norm_name + "_instFlux"
+        output_fluxerr_name = norm_name + "_instFluxErr"
+        output_flag_name = norm_name + "_flag"
+
+        if catalog.isContiguous():
+            catalog[output_flux_name] = catalog[input_flux_name] * corrections
+            catalog[output_fluxerr_name] = catalog[input_fluxerr_name] * corrections
+
+            output_flag = catalog[input_flag_name].copy()
+            output_flag[corrections <= 0.0] = True
+            catalog[output_flag_name] = output_flag
+        else:
+            # If the catalog is not contiguous we must go row-by-row.
+            for i, row in enumerate(catalog):
+                row[output_flux_name] = row[input_flux_name] * corrections[i]
+                row[output_fluxerr_name] = row[input_fluxerr_name] * corrections[i]
+
+                if row[input_flag_name] or corrections[i] <= 0.0:
+                    row[output_flag_name] = True
+
+        ap_corr_map = ApCorrMap()
+        ap_corr_map[raw_flux_name] = ap_corr_field
+        ap_corr_map[raw_fluxerr_name] = ap_corr_err_field
+
+        return lsst.pipe.base.Struct(
+            ap_corr_map=ap_corr_map,
+        )
+
+    def _measure_aperture_correction(self, exposure, catalog):
+        """Internal method to do the aperture correction measurement.
+
+        This measures the aperture correction with the regular task,
+        and if that fails does a fallback median estimate.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure the normalized calibration flux is measured on.
+            This is only used for the bounding box.
```

Comment:
For an internal method, it seems like just passing the `bbox` directly without the `exposure` should do.

Reply:
Actually, is this true, or at least limited to this method? It looks like this gets passed on to `measure_ap_corr` and this docstring shouldn't comment what `measure_ap_corr` needs from the `exposure`.